### DISCUSSION TITLE
Feature Request: Focus on "key" field in newly added tag

### DISCUSSION THREAD

**Discussion by OpenOrangeData** [osmandapp/OsmAnd#20311] [Ideas]
I've been using (and recommending) OsmAnd+ for years now and although for beginners some aspects may be hard to grasp, I'm really happy with most of its features - thanks for the great work!

The following is related to the **Plugin "OpenStreetMap editing"**:
Recently, I've been editing OSM very often through OsmAnd+. It works well for keys/values that I already know by heart, however, what would really help to speed up editing would be if the cursor would automatically "jump" (focus) to the "key" field once I hit the button "add tag". 
Currently, an empty tag is added and automatically scrolled to, so that the user/editor can see it (so far so good), but the cursor is stuck in the field which was edited last (unwanted behaviour) instead of letting me directly type the key of the new tag.

Currently I often think I'm already in the new key and continue typing, only to discover that I'm actually adding letters to an old tag which
makes it invalid. E.g. I already marked a bicycle parking, then I want to add "covered = yes", but instead of writing "covered" in the new tag, it turns out I'm accidentally producing the value "Bicycle parking**covered**".

Is this feature (from my perspective almost a bug ;-)) something you're already working on?

**Comment by OpenOrangeData** [comment]
It seems that with the last release, this was introduced exactly as I wanted: Adding a new tag puts the focus/cursor in the "key" field of the new tag.

However, unfortunately, now I have to manually tap on the "value" field, hitting "next" on my phone's keyboard doesn't have any effect. Could this be fixed?

**Comment by yuriiurshuliak** [comment]
The corresponding task has been created here: [GitHub Issue #20896](https://github.com/osmandapp/OsmAnd/issues/20896).

Your feature request will be considered for future updates. Thank you for contributing to the app!

**Comment by yuriiurshuliak** [comment]
The bug was not reproduced. To reproduce  the issue, try the following steps: click on any POI, choose "Actions," then select "Create POI" or "Modify POI." Next, go to the "Advanced" tab and click "Add tag."

In my case, the cursor automatically focuses on the selected field, and everything functions as expected.

@OpenOrangeData , could you please share a video showing the issue from your side?

OsmAnd~ 4.9.0#3300m, released: 2024-09-27

https://github.com/user-attachments/assets/d950dfbc-6422-493f-9c03-73c8e79e52e7

**Reply by OpenOrangeData** [reply]
@yuriiurshuliak I think you tried this out based on my original description, correct? However, it turned out that in one of the latest releases this problem was fixed, which I commented on 2024-08-05. (In order to make it easier to follow, I have made a separate comment on instead of editing the original description).
As I mentioned in my comment, there's still a problem: tapping on the "next" button doesn't change from the "key" field to the "value" field. (I guess in the case of your video it would be the arrow key **-->|** on the bottom right, which is not labelled "next".) Could you please change this behaviour?

**Reply by yuriiurshuliak** [reply]
We appreciate your feedback and will consider this for future updates.
